services/convert/convert.go

```diff
@@ -105,33 +105,60 @@ func ToBranch(ctx context.Context, repo *repo_model.Repository, branchName strin
 	return branch, nil
 }
 
-// ToBranchProtection convert a ProtectedBranch to api.BranchProtection
-func ToBranchProtection(ctx context.Context, bp *git_model.ProtectedBranch) *api.BranchProtection {
-	pushWhitelistUsernames, err := user_model.GetUserNamesByIDs(ctx, bp.WhitelistUserIDs)
-	if err != nil {
-		log.Error("GetUserNamesByIDs (WhitelistUserIDs): %v", err)
-	}
-	mergeWhitelistUsernames, err := user_model.GetUserNamesByIDs(ctx, bp.MergeWhitelistUserIDs)
-	if err != nil {
-		log.Error("GetUserNamesByIDs (MergeWhitelistUserIDs): %v", err)
+// getWhitelistEntities returns the names of the entities that are in the whitelist
+func getWhitelistEntities(entities []any, whitelistIDs []int64) []string {
+	whitelistIDsMap := make(map[int64]struct{})
```

Comment:
`container.Set` is for this case.

Reply:
https://pkg.go.dev/container

Reply:
It's `code.gitea.io/gitea/modules/container` not that one.